gdk/common/exceptions/BuildError.py

```diff
@@ -0,0 +1,28 @@
+class BuildException(Exception):
```

Comment:
Whats the idea behind creating internal exception classes? I understand it helps in testing but we should have guideline on when to create a new exception class, otherwise this practice could be overused.

Reply:
It not only segregates the error messages based on command type and provide command specific help, it also provides better formatting removing redundant error strings by catching all of them at top level. For example, the solution provided in some of these exceptions are specific to the build command. 
 
I totally agree with the problem of overusing them. But if we expect a possible exception and want to provide customers with very a specific solution or hint for it or even exit, we'd want to create  an exception for it.

Reply:
As per our offline discussion, we agree to add new exception if we want client to handle it differently. Lets follow this as a practice